pkg/kube/cluster-init.sh

```diff
@@ -155,6 +158,34 @@ check_overwrite_nsmounter() {
         ### REMOVE ME-
 }
 
+check_start_k3s() {
+  pgrep -f "k3s server" > /dev/null 2>&1
+  if [ $? -eq 1 ]; then
+      if [ $RESTART_COUNT -lt $MAX_K3S_RESTARTS ]; then
+          ## Must be after reboot, or from k3s restart
+          RESTART_COUNT=$((RESTART_COUNT+1))
+          ln -s /var/lib/k3s/bin/* /usr/bin
+          logmsg "Starting k3s server, restart count: $RESTART_COUNT"
+          # for now, always copy to get the latest
+          nohup /usr/bin/k3s server --config /etc/rancher/k3s/config.yaml &
+          k3s_pid=$!
+          # Give the embedded etcd in k3s priority over io as its fsync latencies are critical
+          ionice -c2 -n0 -p $k3s_pid
+          # Default location where clients will look for config
+          # There is a very small window where this file is not available
+          # while k3s is starting up
+          while [ ! -f /etc/rancher/k3s/k3s.yaml ]; do
+            sleep 5
+          done
+          ln -s /etc/rancher/k3s/k3s.yaml ~/.kube/config
+          cp /etc/rancher/k3s/k3s.yaml /run/.kube/k3s/k3s.yaml
+          sleep 10
```

Comment:
Seems ad-hoc to sleep here. Should the script wait for something it can explicitly check?

Reply:
I can change this to check k3s status instead

Reply:
moved the sleep out of check_start_k3s, caller will sleep to be nicer to the api server while polling the node ready state.